### PR TITLE
DCS-481 Updates

### PR DESCRIPTION
**Pull request by andrewrlee**
* Update dependencies
* Move to validate tokens using jwk
* Move to dps gradle plugin
* Move to use dps orb

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,103 +1,28 @@
...
+  owasp: entur/owasp@0.0.10
+  hmpps: ministryofjustice/hmpps@1.0.0
+
 executors:
-  deployer:
-    docker:
-      - image: circleci/python:3
-    working_directory: ~/app
-  builder:  
+  validator:
```
Comment (petergphillips) [inline .circleci/config.yml:8]: can use the `hmpps\java` executor instead - has all the configuration you've specified here -https://github.com/ministryofjustice/hmpps-circleci-orb/blob/master/src/executors/java.yml

File: src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt [hunk 2]
```diff
@@ -65,6 +67,7 @@ class ClientTrackingTelemetryModuleTest(
         assertThat(insightTelemetry["clientId"]).isEqualTo("elite2apiclient")
     }
 
+    @Disabled("Expiry date no longer validated - similar to whereabouts")
```
Comment (andrewrlee) [inline src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt:70]: Need to check that this is ok..

File: build.gradle.kts [hunk 3]
```diff
@@ -1,101 +1,45 @@
...
 
 configurations {
-
-    implementation {
-        exclude(mapOf("module" to "tomcat-jdbc"))
-    }
+    implementation { exclude(mapOf("module" to "tomcat-jdbc")) }
 }
 
+ext ["spring-security.version"] = "5.3.1.RELEASE"
```
Comment (petergphillips) [inline build.gradle.kts:11]: latest is 5.3.2

File: build.gradle.kts [hunk 4]
```diff
@@ -1,101 +1,45 @@
...
     runtimeOnly("com.h2database:h2:1.4.200")
-    runtimeOnly("org.flywaydb:flyway-core:6.3.3")
+    runtimeOnly("org.flywaydb:flyway-core:6.4.1")
     runtimeOnly("org.postgresql:postgresql:42.2.12")
 
     implementation(kotlin("stdlib-jdk8"))
     implementation(kotlin("reflect"))
 
-    implementation("org.springframework.boot:spring-boot-starter-web")
     implementation("org.hibernate:hibernate-java8")
```
Comment (petergphillips) [inline build.gradle.kts:24]: don't need hibernate as specifying `data-jpa`, but if do want to specify then use `hibernate-core` instead

File: build.gradle.kts [hunk 5]
```diff
@@ -1,101 +1,45 @@
...
 dependencies {
 
     annotationProcessor("org.springframework.boot:spring-boot-configuration-processor")
 
     runtimeOnly("com.h2database:h2:1.4.200")
-    runtimeOnly("org.flywaydb:flyway-core:6.3.3")
+    runtimeOnly("org.flywaydb:flyway-core:6.4.1")
     runtimeOnly("org.postgresql:postgresql:42.2.12")
 
     implementation(kotlin("stdlib-jdk8"))
```
Comment (petergphillips) [inline build.gradle.kts:21]: these two are included in plugin

File: build.gradle.kts [hunk 6]
```diff
@@ -1,101 +1,45 @@
...
-    implementation("org.springframework.boot:spring-boot-starter-actuator")
     implementation("org.springframework.boot:spring-boot-starter-data-jpa")
     implementation("org.springframework.boot:spring-boot-starter-cache")
     implementation("org.springframework.boot:spring-boot-starter-security")
-    implementation("org.springframework.security.oauth:spring-security-oauth2:2.4.0.RELEASE")
-    implementation("org.springframework.security:spring-security-jwt:1.1.0.RELEASE")
+    implementation("org.springframework.boot:spring-boot-starter-oauth2-resource-server")
+    implementation("org.springframework.boot:spring-boot-starter-oauth2-client")
+
+    implementation("org.springframework.security:spring-security-oauth2-jose")
```
Comment (petergphillips) [inline build.gradle.kts:31]: don't need `jose` if specifying `client`

File: build.gradle.kts [hunk 7]
```diff
@@ -1,101 +1,45 @@
...
+
     implementation("net.logstash.logback:logstash-logback-encoder:6.3")
```
Comment (petergphillips) [inline build.gradle.kts:33]: don't need `logback` as included in plugin

File: src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt (hunk 2 above)
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt:70]: test should just be deleted

File: src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt [hunk 8]
```diff
@@ -3,8 +3,10 @@ package uk.gov.justice.hmpps.probationteams.config
 import com.microsoft.applicationinsights.web.internal.RequestTelemetryContext
 import com.microsoft.applicationinsights.web.internal.ThreadContext
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.Ignore
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt:6]: random junit 4 import?

File: src/main/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModule.kt [hunk 9]
```diff
@@ -1,33 +1,24 @@
-package uk.gov.justice.hmpps.probationteams.config
+package uk.gov.justice.digital.hmpps.probationteams.config
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModule.kt:1]: package change to `digital` - rest don't seem to be under digital

File: helm_deploy/probation-teams/templates/_envs.tpl [hunk 10]
```diff
@@ -22,6 +22,9 @@ env:
         name: {{ template "app.name" . }}
         key: APPINSIGHTS_INSTRUMENTATIONKEY
 
+  - name: OAUTH_ENDPOINT_URL
```
Comment (petergphillips) [inline helm_deploy/probation-teams/templates/_envs.tpl:25]: need to remove `JWT_PUBLIC_KEY` from this file too?

File: .circleci/config.yml (hunk 1 above)
Comment (andrewrlee) [reply .circleci/config.yml]: moved to use this and removed old executor

File: src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt (hunk 2 above)
Comment (andrewrlee) [reply src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt]: deleted

File: build.gradle.kts (hunk 3 above)
Comment (andrewrlee) [reply build.gradle.kts]: updated

File: build.gradle.kts (hunk 4 above)
Comment (andrewrlee) [reply build.gradle.kts]: just relying on data jpa instead

File: build.gradle.kts (hunk 5 above)
Comment (andrewrlee) [reply build.gradle.kts]: removed

File: build.gradle.kts (hunk 6 above)
Comment (andrewrlee) [reply build.gradle.kts]: removed

File: build.gradle.kts (hunk 7 above)
Comment (andrewrlee) [reply build.gradle.kts]: removed

File: src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt (hunk 2 above)
Comment (andrewrlee) [reply src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt]: deleted

File: src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt (hunk 8 above)
Comment (andrewrlee) [reply src/test/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModuleTest.kt]: deleted

File: src/main/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModule.kt (hunk 9 above)
Comment (andrewrlee) [reply src/main/java/uk/gov/justice/hmpps/probationteams/config/ClientTrackingTelemetryModule.kt]: moved this

File: helm_deploy/probation-teams/templates/_envs.tpl (hunk 10 above)
Comment (andrewrlee) [reply helm_deploy/probation-teams/templates/_envs.tpl:22]: got rid

File: .circleci/config.yml [hunk 11]
```diff
@@ -1,103 +1,20 @@
...
-            rm BOOT-INF/classes/META-INF/build-info.properties
-
-            # make up a good version name
-            TAG=${APP_VERSION}.${CIRCLE_BRANCH}
-            if [[ "${CIRCLE_BRANCH}" == "master" ]]; then
-              TAG=${APP_VERSION}
-            fi
-            echo $TAG > ./app_version.txt
-
+          command: ./gradlew test
```
Comment (petergphillips) [inline .circleci/config.yml:17]: switching to `test` instead means you need the multi stage Dockerfile changes too

**Pull request merged**